torch/csrc/jit/python/pybind_utils.cpp

```diff
@@ -52,16 +52,29 @@ IValue toIValue(py::handle obj, const TypePtr& type, c10::optional<int32_t> N) {
       auto tuple_type = type->cast<TupleType>();
       const auto& elem_types = tuple_type->elements();
       if (elem_types.size() != tuple_size) {
-        throw py::cast_error(c10::str(
-            "Object ",
-            py::str(obj),
-            " had a different number of elements than type ",
-            type->repr_str()));
+        if (!(tuple_size == 0 && type->repr_str() == "Tuple[None]")) {
```

Comment:
Checking equality of `repr_str` is brittle - I suggest creating a `TupleType` instance with `NoneType` as the only element and using `TupleType::operator==` instead.